src/main/java/com/minecolonies/tileentities/TileEntityColonyBuilding.java

```diff
@@ -135,16 +115,63 @@ public Colony getColony()
         }
         return colony;
     }    @Override
-    public void writeToNBT(NBTTagCompound compound)
+    public void update()
     {
-        super.writeToNBT(compound);
-        if (colonyId == 0)
+        super.update();
+
+        if (!worldObj.isRemote && colonyId == 0)
         {
-            throw new IllegalStateException(String.format("TileEntityColonyBuilding at %s:[%d,%d,%d] has no colonyId; %s colony reference.",
-              worldObj.getWorldInfo().getWorldName(), pos.getX(), pos.getY(), pos.getZ(),
-              colony == null ? "NO" : "valid"));
+            final Colony tempColony = ColonyManager.getColony(worldObj, this.getPosition());
+            if (tempColony != null)
+            {
+                colonyId = tempColony.getID();
+            }
+        }
+    }
+
+    /**
+     * Synchronises colony references from the tile entity
+     */
+    private void updateColonyReferences()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "updateColonyReferences" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)